configd/src/apps/sentinel/report-connectivity.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "report-connectivity.h"
+#include "connectivity.h"
+#include <vespa/config/common/exceptions.h>
+#include <vespa/log/log.h>
+#include <chrono>
+
+LOG_SETUP(".report-connectivity");
+
+using cloud::config::ModelConfig;
+using namespace std::chrono_literals;
+
+namespace config::sentinel {
+
+SinglePing::~SinglePing() = default;
+
+void SinglePing::returnStatus(bool ok) {
+    status = ok ? "ok" : "ping failed";
+    LOG(debug, "peer %s [port %d] -> %s", peerName.c_str(), peerPort, status.c_str());
+    parent.requestDone();
+}
+
+void SinglePing::startCheck(FRT_Supervisor &orb) {
+    check = std::make_unique<PeerCheck>(*this, peerName, peerPort, orb, 2500);
+}
+
+ReportConnectivity::ReportConnectivity(FRT_RPCRequest *req, FRT_Supervisor &orb)
+  : _parentRequest(req),
+    _orb(orb),
+    _result(),
+    _configFetcher()
+{
+    try {
+        _configFetcher.subscribe<ModelConfig>("admin/model", this, 2000ms);
+        _configFetcher.start();
+        return;
+    } catch (ConfigTimeoutException & ex) {
```

Comment:
we should not block the network thread; consider a separate service/thread that keeps the model config updated that we can get a snapshot from.